aws_lambda_powertools/metrics/metrics.py

```diff
@@ -71,15 +71,15 @@ def do_something():
         When metric object fails EMF schema validation
     """
 
-    _metrics = {}
-    _dimensions = {}
-    _metadata = {}
+    _metrics: Dict[str, Any] = {}
```

Comment:
This is possible the most complex, so happy to leave as Any for now until there is a cleaner way.

For example, it might deserve a generic like MetricsContainer, since at the end this holds:

```
>>> my_metric._metrics
{'metric_one': defaultdict(<class 'list'>, {'Unit': 'Count', 'Value': [1.0]}), 'metric_two': defaultdict(<class 'list'>, {'Unit': 'Count', 'Value': [1.0]})}
```

It's a nested dict, where the dictionary is a default dict with `Unit` holding a `str`, and `Value` holding a `List[Number]`

Reply:
Yep, this is kind of where typing falls apart, but also it does highlight the inherit complexity too.